core/src/main/java/org/bitcoinj/script/ScriptChunk.java

```diff
@@ -108,33 +108,37 @@ public boolean isShortestPossiblePushData() {
         return opcode == OP_PUSHDATA4;
     }
 
-    public void write(OutputStream stream) throws IOException {
+    public byte[] toByteArray() {
         if (isOpCode()) {
             checkState(data == null);
-            stream.write(opcode);
+            return new byte[]{ (byte) opcode };
         } else if (data != null) {
+            ArrayList<Byte> res = new ArrayList<>();
             if (opcode < OP_PUSHDATA1) {
                 checkState(data.length == opcode);
-                stream.write(opcode);
+                res.add((byte) opcode);
             } else if (opcode == OP_PUSHDATA1) {
                 checkState(data.length <= 0xFF);
-                stream.write(OP_PUSHDATA1);
-                stream.write(data.length);
+                res.add((byte) OP_PUSHDATA1);
+                res.add((byte) data.length);
             } else if (opcode == OP_PUSHDATA2) {
                 checkState(data.length <= 0xFFFF);
-                stream.write(OP_PUSHDATA2);
-                stream.write(0xFF & data.length);
-                stream.write(0xFF & (data.length >> 8));
+                res.add((byte) OP_PUSHDATA2);
+                res.add((byte) (0xFF & data.length));
+                res.add((byte) (0xFF & (data.length >> 8)));
             } else if (opcode == OP_PUSHDATA4) {
                 checkState(data.length <= Script.MAX_SCRIPT_ELEMENT_SIZE);
-                stream.write(OP_PUSHDATA4);
-                Utils.uint32ToByteStreamLE(data.length, stream);
+                res.add((byte) OP_PUSHDATA4);
```

Comment:
I think you should be using `uint32ToByteArrayLE()` here. Kind of inlining the logic from the method is a step back imho.

Reply:
The drawback of using `uint32ToByteArrayLE` is that a new array must be created and passed. I preferred to keep the code more readable inlining the logic, since the same logic is already done with the case of PUSHDATA2.

Reply:
I took the opportunity and migrated all remaining manual bitstuffing to Utils. See #1518.

If we stay with ByteBuffer (it was just a random idea out of my head), then probably we should add the appropriate utils methods too. Bitstuffing has been implemented incorrectly many times in the past, which is why I'm resitant to duplicate it in code.